server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -686,13 +689,50 @@ public String toString() {
             return "tsid";
         }
 
+        /**
+         * @param value The TSID as a {@link BytesRef}
+         * @return the Base 64 encoded TSID
+         */
         @Override
         public Object format(BytesRef value) {
-            return TimeSeriesIdFieldMapper.decodeTsid(new BytesArray(value).streamInput());
+            // NOTE: we need to do this copy to avoid encoding spurious bytes at the end of the
+            // BytesRef buffer. When sorting results the TermOrdValComparator will copy
+            // values using a BytesRefBuilder whose underlying bytes buffer is sized incorrectly,
+            // allocating more bytes than actually required. As a result of these additional bytes
+            // the Base 64 encoding might include spurious bytes (typically 0s) which result in
+            // additional unwanted TSID trailing characters.
+            try {
+                // NOTE: if the tsid is a map of dimension key/value pairs (as it was before introducing
+                // tsid hashing) we just decode the map and return it.
+                return TimeSeriesIdFieldMapper.decodeTsidAsMap(value);
```

Comment:
Is this to reduce the number changes needed in tests? Because formatting the _tsid field value this way only makes sense for older indices? At some point there won't be any of these indices and we still format _tsid field values this way.

Reply:
This is to support cases in which we have, for instance, a datastream that includes mixed indices...If we don't use he time series aggregator (for instance we use a terms agg on _tsid) and the _tsid field is not stored as a map `decodeTsidAsMap` wouls fail because there is no time series aggregator in the execution path that does the adjustment recreating the map of key/value pairs. Does it make sense?